### ISSUE TITLE
Wrong deletion of moderation requests after accepting or refusing

### ISSUE THREAD

**Issue by maierthomas** [eclipse/sw360#143]
<a href="https://github.com/ChristophNiehoff"><img src="https://avatars1.githubusercontent.com/u/10673021?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [ChristophNiehoff](https://github.com/ChristophNiehoff)**
_Thursday Sep 07, 2017 at 15:05 GMT_
_Originally opened as https://github.com/sw360/sw360portal/issues/567_

----

An error is thrown after accepting a delete moderation request (for projects, components and releases).

The reason is the following:

After commit [`6acac74e`](https://github.com/sw360/sw360portal/commit/6acac74ed584dd56d5bf1ad7d72f20847941a67d), `ModerationPortlet.renderActionView` now contains the lines
```
                } else if (ACTION_ACCEPT.equals(action)) {
                    String requestingUserEmail = moderationRequest.getRequestingUser();
                    User requestingUser = UserCacheHolder.getUserFromEmail(requestingUserEmail);
                    acceptModerationRequest(user, requestingUser, moderationRequest, request);

                    client.acceptRequest(id, moderationComment, user.getEmail());
```
(and similar for declining a request).

* The function `acceptModerationRequest` (e.g. for accepting a project delete request)  leads to the `ProjectDatabasehandler.removeProjectAndCleanUp`, which removes the project from the database and calls `ProjectModerator.notifyModeratorOnDelete`. But this method does not notify the moderator but rather deletes the moderation request from the database.

* The function `client.acceptRequest` tries to set the `ModerationState` of the request to approved, but fails as the request is already deleted. In the end, this results in a `NullPointerException`.

Long story short, I believe that `ProjectModerator.notifyModeratorOnDelete` should be corrected to not delete the moderation request. Also, the naming of the two "acceptRequest"-functions (and also decline-functions) in `ModerationPortlet.renderActionView` is confusing.

**Comment by maierthomas** [comment]
<a href="https://github.com/ChristophNiehoff"><img src="https://avatars1.githubusercontent.com/u/10673021?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ChristophNiehoff](https://github.com/ChristophNiehoff)**
_Thursday Sep 07, 2017 at 15:05 GMT_

----

@alexbrdn Could you have a look at it?

**Comment by maierthomas** [comment]
<a href="https://github.com/alexbrdn"><img src="https://avatars0.githubusercontent.com/u/16780167?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [alexbrdn](https://github.com/alexbrdn)**
_Thursday Sep 07, 2017 at 15:31 GMT_

----

I guess I have to. What, after introducing all these problems...

**Comment by nutanv1contr** [comment]
Retested the issue and its still reproducible.

**Comment by GMishx** [comment]
@keerthi-bl , can you please help retest this issue?
